native/cocos/3d/assets/Mesh.cpp

```diff
@@ -1071,9 +1091,148 @@ void Mesh::accessAttribute(index_t primitiveIndex, const char *attributeName, co
     }
 }
 
-/* static */
+void Mesh::updateVertexFormat() {
+#if CC_OPTIMIZE_MESH_DATA
+    uint8_t *data = _data.buffer()->getData();
+    ccstd::vector<uint32_t> attributeIndicsNeedConvert;
+
+    for (auto &vertexBundle : _struct.vertexBundles) {
+        // NOTE: Don't use reference here since we need to copy attributes
+        const auto orignalAttributes = vertexBundle.attributes;
+
+        auto &attributes = vertexBundle.attributes;
+        auto &view = vertexBundle.view;
+        uint32_t offset = view.offset;
+        uint32_t length = view.length;
+        uint32_t count = view.count;
+        const uint32_t stride = view.stride;
+        uint32_t dstStride = stride;
+
+        CC_ASSERT(count * stride == length);
+
+        attributeIndicsNeedConvert.clear();
+        attributeIndicsNeedConvert.reserve(orignalAttributes.size());
+
+        uint32_t attributeIndex = 0;
+        for (auto &attribute : attributes) {
+            if (attribute.name == gfx::ATTR_NAME_NORMAL) {
+                if (attribute.format == gfx::Format::RGB32F) {
+                    attributeIndicsNeedConvert.emplace_back(attributeIndex);
+                    attribute.format = gfx::Format::RGB16F;
+    #if (CC_PLATFORM == CC_PLATFORM_IOS) || (CC_PLATFORM == CC_PLATFORM_MACOS)
+                    dstStride -= 4; // NOTE: Metal needs 4 bytes alignment
+    #else
+                    dstStride -= 6;
+    #endif
+                }
+            } else if (attribute.name == gfx::ATTR_NAME_TEX_COORD || attribute.name == gfx::ATTR_NAME_TEX_COORD1 || attribute.name == gfx::ATTR_NAME_TEX_COORD2 || attribute.name == gfx::ATTR_NAME_TEX_COORD3 || attribute.name == gfx::ATTR_NAME_TEX_COORD4 || attribute.name == gfx::ATTR_NAME_TEX_COORD5 || attribute.name == gfx::ATTR_NAME_TEX_COORD6 || attribute.name == gfx::ATTR_NAME_TEX_COORD7 || attribute.name == gfx::ATTR_NAME_TEX_COORD8) {
```

Comment:
Use `switch case` instead? It is too long here.

Reply:
The name is `std::string` type, could not be use in switch case.